tests/iio_stresstest.c

```diff
@@ -385,18 +434,45 @@ int main(int argc, char **argv)
 			break;
 		case 'b':
 			info.arg_index += 2;
-			info.buffer_size = strtol(info.argv[info.arg_index], NULL, 10);
-			/* Max 4M */
-			if (info.buffer_size > (1024 * 1024 * 4))
-				info.buffer_size = 1024 * 1024 * 4;
+			tmp = strtol(info.argv[info.arg_index], NULL, 10);
+			/* Max 4M , min 64 bytes (cache line) */
+			if (tmp > (1024 * 1024 * 4)) {
+				fprintf(stderr, "Limiting buffersize to 4M\n");
+				tmp = 1024 * 1024 * 4;
+			}
+			if (tmp < min_samples) {
+				fprintf(stderr, "Setting buffersize to %zu bytes\n", min_samples);
+				tmp = min_samples;
+			}
+			info.buffer_size = (unsigned int)tmp;
 			break;
 		case 't':
 			info.arg_index +=2;
-			info.timeout = 1000 * strtol(info.argv[info.arg_index], NULL, 10);
+			tmp = strtol(info.argv[info.arg_index], NULL, 10);
```

Comment:
should we use `strtoul()` for all these?
doesn't look like we have values that should be negative yet;

then the `tmp < 0` check goes away;

also, this code block is repeated a few times;
it could be an idea to implement it as a function/routine called  `clamp()`
maybe something like:
```
int clamp(const char *name, val, min, max) {
       if (val > max) {
               val = max;
               fprtinf("Clamped %s to max %u", name, max)
       } 
       if (val < min) {
              val = min;
              fprtinf("Clamped %s to min %u", name, min)
        }
       return val;
}
```

Reply:
fixed